src/View/Helper/FormHelper.php

```diff
@@ -1167,9 +1170,14 @@ protected function _optionsOptions($fieldName, $options)
             return $options;
         }
 
+        if (substr($fieldName, -5) === '._ids') {
+            $fieldName = substr($fieldName, 0, -5);
+        } elseif (substr($fieldName, -3) === '_id') {
+            $fieldName = substr($fieldName, 0, -3);
+        }
         $fieldName = array_slice(explode('.', $fieldName), -1)[0];
         $varName = Inflector::variable(
-            Inflector::pluralize(preg_replace('/_id$/', '', $fieldName))
+            Inflector::pluralize($fieldName)
```

Comment:
Manually changed association properties and foreign keys aside - As per conventions, it should only be necessary to pluralize belongsTo/hasOne fields, shouldn't it? If that is the case, could pluralizing based on the postfix be a viable solution (given that it's worth to keep the magic anyways)?